### ISSUE TITLE
innerHTML for inline scripts tags

### ISSUE THREAD

**Issue by romanonthego** [nfl/react-helmet#65]
hey.
we have been working on the project that requires using schema.org metadata.
it is turned out that most useful and rational way is to use ld+json notation (https://developers.google.com/schemas/formats/json-ld?hl=en)
so wee need not only have support for 
```javascript 
{src: "http://include.com/pathtojs.js", type: "text/javascript"}
```
but also for 
```javascript
{
  name: "company",
  type: "application/ld+json"
  innerHTML: `
    {
      "@context": "http://schema.org",
      "@type": "Organization",
      "url": "${url}",
      "logo": "${requrie('logo.png')}"
    }
  `
}
```

to result in something like:
```html
<script name="company" type="application/ld+json" data-react-helmet="true">
  {
    "@context": "http://schema.org",
     "@type": "Organization",
     "name": "Whitescape",
     "url": "http://site.com",
    "logo": "http://site.com/assets/logo@166f4fc0174286c55eaef4a17eea9cc8.png"
  }
</script>
```
I already have working solution in our company fork, so if you are interesting i'll finish this, add some test and send a pull request.

p.s.
yes it will allow to use a inline scripts in Helmet and possibly shoot yourself in the foot, but since it is a "A document head manager for React" Helmet should cover everything you may want in document head, is it not?

**Comment by cwelch5** [comment]
This sounds fine.  I agree, it's up to users to not trip themselves up.  Once you add unit tests, and figure out how to still escape values for non-script tags, then please submit a PR.  Thanks!

**Comment by yannickcr** [comment]
I have the same need for my team, but for now this feature is still missing in `react-helmet`.

Are you still working on it @romanonthego? If not (and if @cwelch5 is still ok), I'll be happy to submit a PR for this.

**Comment by romanonthego** [comment]
oh yeh, my bad.
i am totally gonna finish this one as soon as I'll get through current tasks (next week probably)
for now @yannickcr you could use https://github.com/whitescape/react-helmet/ - it's our fork, running in production in couple of projects without problems for now but with obvious lack of test.

**Comment by ericfong** [comment]
+1 better can launch to npm. Thanks

**Comment by MikaFima** [comment]
:+1:

**Comment by shouze** [comment]
:+1:

**Comment by nuc** [comment]
Hey there,

Since @romanonthego's fork was a bit outdated I rebased master on it, cleaned it up a bit and added a simple test case over there:

https://github.com/nfl/react-helmet/compare/master...moviepilot:master

Do you think there's anything else missing?

Otherwise it would be great to have this merged, so we won't need to use our own fork.

Thanks!

**Comment by jks8787** [comment]
super happy this is going to get in :fireworks:

**Comment by romanonthego** [comment]
yeh, sorry about this, closing now

**Comment by cwelch5** [comment]
Fixed with [react-helmet@3.0.0](https://www.npmjs.com/package/react-helmet)
